Bug: Qualified paths to `IUnknown` cause unrelated compiler error with `#[interface]` proc macro

### Which crate is this about?

windows

### Crate version

0.35

### Summary

Using a fully qualified path to `windows::core::IUnknown` with the `#[interface("...")]` proc macro causes a compiler error that doesn't make much sense. This worked previously in 0.34.

### Toolchain version/configuration

Default host: x86_64-pc-windows-msvc
rustup home:  C:\Users\wesleywiser\.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-msvc (default)
beta-x86_64-pc-windows-msvc
nightly-x86_64-pc-windows-msvc
stage1-1
stage1-2
stage1-3
stage1-4

installed targets for active toolchain
--------------------------------------

x86_64-pc-windows-msvc

active toolchain
----------------

nightly-x86_64-pc-windows-msvc (directory override for 'D:\code\temp\windows_rs_com_bug')
rustc 1.62.0-nightly (90ca44752 2022-04-11)


### Reproducible example

```rust
use windows::core::*;

#[interface("ABCDABCD-ABCD-ABCD-ABCD-ABCDABCDABCD")]
pub unsafe trait IFoo: windows::core::IUnknown {
}
```


### Crate manifest

```toml
[package]
name = "windows_rs_com_bug"
version = "0.1.0"
edition = "2021"

[dependencies]
windows = { version = "0.35", features = ["alloc", "implement", "interface" ] }
```


### Expected behavior

`cargo build` completes successfully

### Actual behavior

```
PS D:\code\temp\windows_rs_com_bug> cargo build
   Compiling windows_rs_com_bug v0.1.0 (D:\code\temp\windows_rs_com_bug)
error[E0107]: this associated function takes 2 generic arguments but 3 generic arguments were supplied
  --> src\lib.rs:3:1
   |
3  | #[interface("ABCDABCD-ABCD-ABCD-ABCD-ABCDABCDABCD")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected 2 generic arguments
   | help: remove this generic argument
   |
note: associated function defined here, with 2 generic parameters: `T`, `OFFSET`
  --> C:\Users\wesleywiser\.cargo\registry\src\github.com-1ecc6299db9ec823\windows-0.35.0\src\core\unknown.rs:71:18
   |
71 |     pub const fn new<T: IUnknownImpl, const OFFSET: isize>() -> Self {
   |                  ^^^ -                -------------------
   = note: this error originates in the attribute macro `interface` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0107`.
error: could not compile `windows_rs_com_bug` due to previous error
```

### Additional comments

_No response_

--- a/crates/libs/interface/src/lib.rs
+++ b/crates/libs/interface/src/lib.rs
@@ -284,7 +284,12 @@ impl Interface {
     }
 
     fn parent_is_iunknown(&self) -> bool {
-        self.parent.is_ident("IUnknown")
+        let end = self.parent.segments.last();
+        let end = match end {
+            Some(e) => e,
+            None => return false,
+        };
+        end.ident == "IUnknown"
     }
 
     fn parent_ident(&self) -> &syn::Ident {

--- a/crates/tests/nightly_interface/tests/com.rs
+++ b/crates/tests/nightly_interface/tests/com.rs
@@ -18,7 +18,7 @@ unsafe trait ICustomUri: IUnknown {
 
 /// A custom declaration of implementation of `IPersist`
 #[interface("0000010c-0000-0000-C000-000000000046")]
-unsafe trait ICustomPersist: IUnknown {
+unsafe trait ICustomPersist: windows::core::IUnknown {
     unsafe fn GetClassID(&self, clsid: *mut GUID) -> HRESULT;
 }
 

